server/resp/src/test/java/org/infinispan/server/resp/RespSetCommandsTest.java

```diff
@@ -132,4 +132,28 @@ public void testSinterstore() {
         // Create a List
         assertWrongType(() -> redis.rpush("listleads", "tristan"), () -> redis.sinterstore("destination", "listleads", "william"));
     }
+
+    @Test
+    public void testSmove() {
+        RedisCommands<String, String> redis = redisConnection.sync();
+        String src = "smove-src";
+        String dst = "smove-dst";
+        redis.sadd(src, "1", "2", "3");
+        redis.sadd(dst, "4", "5");
+        assertThat(redis.smove(src,dst,"2")).isTrue();
+        // No smembers yet, use sadd as contains() function
+        assertThat(redis.sadd(dst,"2")).isEqualTo(0L);
+        assertThat(redis.sadd(src,"2")).isEqualTo(1L);
+
+        assertThat(redis.smove(src,dst,"3")).isTrue();
+        assertThat(redis.smove(src,dst,"3")).isFalse();
+
+        String nesrc = "smove-nonexist-src";
+        assertThat(redis.smove(nesrc,dst,"2")).isFalse();
+
+        String nedst = "smove-nonexist-src";
```

Comment:
We can extend this to cover when source == destination.

Reply:
done